packages/builder/src/components/userInterface/ComponentPropertiesPanel.svelte

```diff
@@ -30,26 +31,53 @@
   let selectedCategory = categories[0]
 
   $: components = $store.components
-  $: componentInstance = $store.currentComponentInfo
   $: componentDefinition = $store.components[componentInstance._component]
   $: componentPropDefinition =
     flattenedPanel.find(
       //use for getting controls for each component property
       c => c._component === componentInstance._component
     ) || {}
 
-  $: panelDefinition = componentPropDefinition.properties
-    ? componentPropDefinition.properties[selectedCategory.value]
-    : {}
+  
+  let panelDefinition = {}
+
+  $: {
+    if(componentPropDefinition.properties) {
+      if(selectedCategory.value === "design") {
+        panelDefinition = componentPropDefinition.properties["design"]
+      }else{
+        let panelDef = componentPropDefinition.properties["settings"]
+        if($store.currentFrontEndType === "page") {
+          panelDefinition = [...page,...panelDef]
+        }else if($store.currentFrontEndType === "screen" && $store.currentView !== "component") {
+          panelDefinition = [...screen, ...panelDef]
+        }else {
+          panelDefinition = panelDef
+        }
+      }
+    }
+  }
 
-  // SCREEN PROPS =============================================
-  $: screen_props =
-    $store.currentFrontEndType === "page"
-      ? getProps($store.currentPreviewItem, ["name", "favicon"])
-      : getProps($store.currentPreviewItem, ["name", "description", "route"])
+  let componentInstance = {}
+  $: {
+     if(($store.currentFrontEndType === "screen" || $store.currentFrontEndType === "page") && $store.currentView !== "component") {
+       componentInstance = {...$store.currentPreviewItem, ...$store.currentComponentInfo}
+     }else {
+       componentInstance = $store.currentComponentInfo
+     }
+  }
 
   const onStyleChanged = store.setComponentStyle
-  const onPropChanged = store.setComponentProp
+
+  function onPropChanged(key, value) {
```

Comment:
think you can just do

```js
if($store.currentView !== "component") {
   store.setComponentProp(key, value);
  return;
}
store.editPageOrScreen(key, value)
```

Reply:
`$store.currentView`  seems to linger on "component" when a screen / page is selected after a component. This seems like a bit of bug so I'll dig a bit deeper.

Reply:
@shogunpurple latest commit should do the job